src/code_memory/embeddings.py

```
@@ -0,0 +1,248 @@
+
+    def _periodic_memory_cleanup(self) -> None:
+        """Perform periodic memory cleanup."""
+        if self._embeddings_since_cleanup >= self.memory_cleanup_threshold:
+            if torch.cuda.is_available():
+                torch.cuda.empty_cache()
+            gc.collect()
+            self.stats.record_memory_cleanup()
+            self._embeddings_since_cleanup = 0
+            logger.debug("Performed memory cleanup")
+
+    async def encode_async(self, texts: List[str], batch_size: int = 32) -> np.ndarray:
```

Ruff (0.11.9)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)